pkg/controller/infinispan/resources/operatorconfig/grafana.go

```diff
@@ -0,0 +1,104 @@
+package operatorconfig
+
+import (
+	consts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	kube "github.com/infinispan/infinispan-operator/pkg/kubernetes"
+	grafanav1alpha1 "github.com/integr8ly/grafana-operator/v3/pkg/apis/integreatly/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// reconcileGrafana reconciles grafana object status with the operator configuration settings
+func (r *ReconcileOperatorConfig) reconcileGrafana(config, currentConfig map[string]string, operatorNs string) (*reconcile.Result, error) {
+	grafanaNs := config[grafanaDashboardNamespaceKey]
+
+	// Delete current grafana dashboard if namespace is changes
+	if err := r.deleteDashboardOnKeyChanged(config, currentConfig); err != nil {
+		return &reconcile.Result{}, err
+	}
+	currentConfig[grafanaDashboardNamespaceKey] = ""
+
+	if grafanaNs == "" {
+		return &reconcile.Result{}, nil
+	}
+	// detect the GrafanaDashboard resource type resourceExists on the cluster
+	resourceExists, err := r.Kube.IsGroupVersionSupported(grafanav1alpha1.SchemeGroupVersion.String(), grafanav1alpha1.GrafanaDashboardKind)
+	if err != nil {
+		r.Log.Error(err, "Error checking Grafana support")
+		return &reconcile.Result{Requeue: true}, nil
+	}
+	if !resourceExists {
+		r.Log.Info("Grafana CRD not present - not installing dashboard CR")
+		return &reconcile.Result{RequeueAfter: consts.DefaultWaitOnCreateResource}, nil
+	}
+	infinispanDashboard := emptyDashboard(config)
+	if _, err = controllerutil.CreateOrUpdate(ctx, r.client, infinispanDashboard, func() error {
```

Comment:
CreateOrUpdate should validate for CreationTimestamp and if it IsZero() that's creation and need to invoke `controllerutil.SetControllerReference` otherwise need to update some changed fields if necessary

Reply:
uh right, `SetControllerReference` is only on create. thanks